packages/app-mobile/components/NotesBar.tsx

```diff
@@ -0,0 +1,310 @@
+			},
+			inputGroup: {
+				justifyContent: 'space-between',
+			},
+		};
+
+		return StyleSheet.create(styles);
+	};
+
+	const titleComp = (
+		<View style={[styles().title, styles().horizontalFlex]}>
+			<Icon name='md-document'style={[styles().top, styles().titleIcon]} />
+			<Text style={[styles().top, styles().titleText]}>{_('Notes')}</Text>
+		</View>
+	);
+
+	const dividerComp = (
+		<View style={styles().divider}></View>
+	);
+
+	const handleNotesBarClose = () => {
+		props.dispatch({ type: 'NOTES_BAR_CLOSE' });
+	};
+
+	const closeButtonComp = (
+		<TouchableOpacity onPress={handleNotesBarClose}>
+			<Icon name="close" style={[styles().top, styles().closeIcon]}/>
+		</TouchableOpacity>
+	);
+
+	const renderIconButton = (icon: JSX.Element, onPress: ()=> Promise<void>) => {
+		return (
+			<TouchableOpacity style={styles().button} activeOpacity={0.8} onPress={onPress}>{icon}</TouchableOpacity>
+		);
+	};
+
+
+	const handleNewNote = async (isTodo: boolean) => {
+		let folderId = props.selectedFolderId !== Folder.conflictFolderId() ? props.selectedFolderId : null;
+		if (!folderId) folderId = props.activeFolderId;
+
+		props.dispatch({
+			type: 'NAV_BACK',
+		});
+
+		const newNote = await Note.save({
+			parent_id: folderId,
+			is_todo: isTodo ? 1 : 0,
+		}, { provisional: true });
+
+		props.dispatch({
+			type: 'NAV_GO',
+			routeName: 'Note',
+			noteId: newNote.id,
+		});
+	};
+
+	const addNoteButtonComp = renderIconButton(<Icon name='document-text-outline' style={styles().buttonIcon} />, () => handleNewNote(false));
+	const addTodoButtonComp = renderIconButton(<Icon name='checkbox-outline' style={styles().buttonIcon} />, () => handleNewNote(true));
+
+	const topComp = (
+		<View>
+			<View style={[styles().topContainer, styles().horizontalFlex]}>
+				{titleComp}
+				{closeButtonComp}
+			</View>
+			{dividerComp}
+		</View>
+	);
+
+	// Copied from './screens/search.js' and modified
+	const refreshSearch = async () => {
+		let notes_ = [];
+
+		if (query) {
+			if (props.settings['db.ftsEnabled']) {
+				notes_ = await SearchEngineUtils.notesForQuery(query, true);
+			} else {
+				const p = query.split(' ');
+				const temp = [];
+				for (let i = 0; i < p.length; i++) {
+					const t = p[i].trim();
+					if (!t) continue;
+					temp.push(t);
+				}
+
+				notes_ = await Note.previews(null, {
+					anywherePattern: `*${temp.join('*')}*`,
+				});
+			}
+
+			const parsedQuery = await SearchEngine.instance().parseQuery(query);
+			const highlightedWords = SearchEngine.instance().allParsedQueryTerms(parsedQuery);
+
+			props.dispatch({
+				type: 'SET_HIGHLIGHTED',
+				words: highlightedWords,
+			});
+
+			setNotes(notes_);
+		}
+	};
+
+	const handleQuerySubmit = async () => {
+		const query_ = query.trim();
+
+		if (!query_) {
+			props.dispatch({
+				type: 'SEARCH_QUERY',
+				query: '',
+			});
+		} else {
+			props.dispatch({
+				type: 'SEARCH_QUERY',
+				query: query_,
+			});
+		}
+
+		setQuery(query_);
```

Comment:
```suggestion
		const trimmedQuery = query.trim();

		if (!trimmedQuery) {
			props.dispatch({
				type: 'SEARCH_QUERY',
				query: '',
			});
		} else {
			props.dispatch({
				type: 'SEARCH_QUERY',
				query: trimmedQuery,
			});
		}

		setQuery(trimmedQuery);
```

Feel free to ignore this suggestion! It's just personal preference.

Reply:
Seems better, thanks